// react-input-files/lib/InputFiles.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _inheritsLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/inheritsLoose"));

var React = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactPortal = require("react-portal");

var InputFiles =
/*#__PURE__*/
function (_React$PureComponent) {
  (0, _inheritsLoose2.default)(InputFiles, _React$PureComponent);

  function InputFiles() {
    var _this;

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _React$PureComponent.call.apply(_React$PureComponent, [this].concat(args)) || this;
    _this.input = React.createRef();

    _this.onClick = function (e) {
      e.preventDefault();
      var input = _this.input.current;
      if (input) input.click();
    };

    _this.onChange = function (e) {
      var onChange = _this.props.onChange;
      onChange(e.target.files, e);
    };

    return _this;
  }

  var _proto = InputFiles.prototype;

  _proto.render = function render() {
    var _this$props = this.props,
        children = _this$props.children,
        accept = _this$props.accept,
        style = _this$props.style,
        multiple = _this$props.multiple;
    var input = this.input,
        onClick = this.onClick,
        onChange = this.onChange;
    return React.createElement(React.Fragment, null, React.createElement("div", {
      onClick: onClick,
      onKeyPress: onClick,
      role: "button",
      tabIndex: "0",
      style: (0, _extends2.default)({
        display: 'inline-block'
      }, style)
    }, children), React.createElement(_reactPortal.Portal, null, React.createElement("input", {
      ref: input,
      type: "file",
      accept: accept,
      onChange: onChange,
      style: {
        display: 'none'
      },
      multiple: multiple
    })));
  };

  return InputFiles;
}(React.PureComponent);

InputFiles.defaultProps = {
  accept: 'image/*'
};
process.env.NODE_ENV !== "production" ? InputFiles.propTypes = {
  children: _propTypes.default.node.isRequired,
  onChange: _propTypes.default.func,
  // (files: Array<File>, e: SyntheticEvent<HTMLInputElement>) => void,
  accept: _propTypes.default.string,
  style: _propTypes.default.object,
  // eslint-disable-line
  multiple: _propTypes.default.bool
} : void 0;
var _default = InputFiles;
exports.default = _default;
module.exports = exports.default;
